src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/RazorProjectHostBase.cs

```diff
@@ -152,6 +181,7 @@ private ProjectSnapshotManagerBase GetProjectManager()
             if (_projectManager == null)
             {
                 _projectManager = (ProjectSnapshotManagerBase)_workspace.Services.GetLanguageServices(RazorLanguage.Name).GetRequiredService<ProjectSnapshotManager>();
+                _projectManager.Changed += ProjectManager_Changed;
```

Comment:
You'll need to put this logic into a different class. There's 1 `RazorProjectHostBase` per-project and there's 1 `ProjectSnapshotManager` per IDE. This means if you have 4 projects you'll be serializing the same project 4x.